### DISCUSSION TITLE
Expose tailwind.config.js

### DISCUSSION THREAD

**Discussion by andrew-culver** [retypeapp/retype#221] [Ideas]
I suggest allowing users to add the `tailwind.config.js` file at some predetermined place in the tree. [Read about it here](https://tailwindcss.com/docs/configuration).

A nice touch would be to merge whatever we put in there with your back-end version, so we only have to indicate what is to be modified.

This covers the basics of customization for me, puts retype in sync with future tailwind customization features, and is well documented so you don't have to.

Andrew Culver

**Comment by sboklund** [comment]
This would be absolutely fantastic!

**Comment by geoffreymcgill** [comment]
Hi @andrew-culver. Thanks for the suggestion. This is going to be a tough one for us to implement.

Currently, there are two build steps for the Retype App distributable:

1. Build the client-side resources, including .js and .css files
2. Build the Retype App, which embeds the client resources within

The first step is where Tailwind is built. That entire client build step can take minutes to build.

There is of course a third _build_ step, and that is to build your website using the Retype App. That build step takes seconds, at most.

To expose the `tailwind.config.js` and rebuild the .css would mean somehow packaging Node and the Tailwind build process inside Retype App, which it currently is not. The size and complexity of the Retype App would bloat up. 

Maybe we could build another service which would accept your `tailwind.config.js` file and would output a custom .css file for your project. Then that compiled .css file, if found by Retype in your project, would be used instead of the embedded default `retype.css` file.

There is nothing in particular that prevents a custom .css file for your project from being used by `retype build`. The trick is more about how to build that custom .css file, efficiently.

It's an interesting problem. We have a team meeting today and will discuss this topic in detail.

**Comment by andrew-culver** [comment]
Now that I know more about how `tailwind.config.js` is implemented, I think it's more of a problem than a solution.

I'm perfectly happy to just edit a `*.css` file if you will put it to work for me. You define what is allowed in it, ignore what is not allowed, and it is up to me to comply. Inasmuch as I do comply, I get the look I want.

If you want to be nice to me, you could run a custom "retypelint" on my edited css to let me know if and where I screwed up. That is, if including that in your client app is not too burdensome.
